src/ray/gcs/client_test.cc

```diff
@@ -451,48 +596,45 @@ void TaskLookupAfterUpdateFailure(gcs::AsyncGcsClient *client, const TaskID &id)
 
 void TestLogSubscribeAll(const JobID &job_id,
```

Comment:
It probably makes sense to also add a `TestSetSubscribeAll` test to make sure that we get the right notification modes.

Reply:
You are right. I added `TestSetSubscribeAll`, `TestSetSubscribeId`, `TestSetSubscribeCancel`. In `TestSetSubscribeAll`, I called `Add` and `Remove` several times with the same data entry to test functionality of adding duplicate entry and removing non-existent entry.

Reply:
Thanks, the tests look good!